docs/router/configuration.mdx

```diff
@@ -1137,13 +1138,25 @@ Legacy WebSocket clients that use the Absinthe protocol might not be able to sen
 |WEBSOCKETS\_ABSINTHE\_ENABLED |enabled| <Icon icon="square" /> |  | true |
 | WEBSOCKETS_ABSINTHE_HANDLER_PATH |handler\_path | <Icon icon="square" /> |The path to mount the Absinthe handler on| /absinthe/socket |
 
-
 ### WebSocket Authentication
 
 It's possible that Authentication for a WebSocket connection is not possible at the HTTP layer. In such a case, you can enable Authentication "from\_initial\_payload". This will extract a value from the "initial\_payload" field in the first WebSocket message which is responsible for negotiating the protocol between client and server.
 
 In addition, it's possible to export the extracted value into a Request Header, which allows the Router to propagate it using [Header Propagation Rules](/router/configuration#global-header-rules) in subsequent Subgraph Requests.
 
+### WebSocket Client Info Configuration
+
+Configure how to extract client info from the initial WebSocket payload. This allows you to set client name and version information from the initial connection payload, which can then be forwarded to request headers if needed.
+
+|Environment Variable|YAML|Required|Description|Default Value|
+|---|---|---|---|---|
+| WEBSOCKETS\_CLIENT\_INFO\_FROM\_INITIAL\_PAYLOAD\_ENABLED |enabled| <Icon icon="square" /> |Enable extracting client info from initial payload| true |
+| WEBSOCKETS\_CLIENT\_INFO\_FROM\_INITIAL\_PAYLOAD\_NAME\_FIELD |name\_field| <Icon icon="square" /> |The field name in the initial payload to extract the client name from|  |
+| WEBSOCKETS\_CLIENT\_INFO\_FROM\_INITIAL\_PAYLOAD\_VERSION\_FIELD |version\_field| <Icon icon="square" /> |The field name in the initial payload to extract the client version from|  |
+| WEBSOCKETS\_CLIENT\_INFO\_FROM\_INITIAL\_PAYLOAD\_FORWARD\_TO\_REQUEST\_HEADERS\_ENABLED |forward\_to\_request\_headers.enabled| <Icon icon="square" /> |Enable forwarding extracted client info to request headers| true |
+| WEBSOCKETS\_CLIENT\_INFO\_FROM\_INITIAL\_PAYLOAD\_NAME\_TARGET\_HEADER |forward\_to\_request\_headers.name\_target\_header| <Icon icon="square" /> |The header name to forward the client name to|  |
+| WEBSOCKETS\_CLIENT\_INFO\_FROM\_INITIAL\_PAYLOAD\_VERSION\_TARGET\_HEADER |forward\_to\_request\_headers.version\_target\_header| <Icon icon="square" /> |The header name to forward the client version to|  |
```

Comment:
Why not put these in backticks for monospace instead of escaping all the underscores?

Reply:
we always did it like this? :D